components/common/link/NavigationLink.tsx

```diff
@@ -0,0 +1,47 @@
+import { type FC, useEffect, useState } from 'react';
+
+import NextLink, { LinkProps as NextLinkProps } from 'next/link';
+import { useRouter } from 'next/router';
+
+import MuiLink, { LinkProps } from '@mui/material/Link';
+
+type NavigationLinkProps = LinkProps & NextLinkProps;
+
+const NavigationLink: FC<NavigationLinkProps> = props => {
+  const { variant, href, children, ...restProps } = props;
+  const { asPath, isReady } = useRouter();
+  const [computedClassName, setComputedClassName] = useState('');
+
+  useEffect(() => {
+    // Check if the router fields are updated client-side
+    if (isReady) {
+      // Dynamic route will be matched via props.as
+      // Static route will be matched via props.href
+      const linkPathname = new URL((props.as || props.href) as string, location.href).pathname;
```

Comment:
As far as I see, `href` has been destructured before. We can safely replace `props.href` with `href`. The same can be done with `props.as` but to add `as` to destructuring. What do you think?